hazelcast/src/test/java/com/hazelcast/cache/CacheCreationTest.java

```diff
@@ -0,0 +1,120 @@
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(SlowTest.class)
+public class CacheCreationTest extends HazelcastTestSupport {
+
+    static Config hzConfig;
+    private static final int THREAD_COUNT = 4;
+
+    @BeforeClass
+    public static void init() throws Exception{
+        final URL configUrl1 = CacheCreationTest.class.getClassLoader().getResource("test-hazelcast-real-jcache.xml");
+        XmlConfigBuilder configBuilder = new XmlConfigBuilder(configUrl1.getFile());
+        hzConfig = configBuilder.build();
+    }
+
+    @Before
+    @After
+    public void killAllHazelcastInstances() throws Exception {
+        Hazelcast.shutdownAll();
+    }
+
+    @Test
+    public void creatingASingleCache() throws URISyntaxException {
+        HazelcastServerCachingProvider cachingProvider = createCachingProvider(hzConfig);
+        Cache<Object, Object> cache = cachingProvider.getCacheManager().getCache("xmlCache" + 1);
+        cache.get(1);
+    }
+
+    @Test
+    public void creatingASingleCacheFromMultiProviders() throws URISyntaxException, InterruptedException {
+
+        ExecutorService execSer = Executors.newFixedThreadPool(THREAD_COUNT);
+
+        final CountDownLatch latch = new CountDownLatch(THREAD_COUNT);
+        for (int i = 0; i < THREAD_COUNT; i++) {
+            execSer.execute(new Runnable() {
+                @Override
+                public void run() {
+                    HazelcastServerCachingProvider cachingProvider = createCachingProvider(hzConfig);
+                    Cache<Object, Object> cache = cachingProvider.getCacheManager().getCache("xmlCache");
+                    cache.get(1);
+                    latch.countDown();
+                }
+            });
+        }
+        HazelcastTestSupport.assertOpenEventually(latch);
+        if (!execSer.awaitTermination(3, TimeUnit.SECONDS)) {
+            execSer.shutdown();
+        }
+    }
+
+    @Test
+    public void creatingMultiCacheFromMultiProviders() throws URISyntaxException, InterruptedException {
+
+        ExecutorService execSer = Executors.newFixedThreadPool(THREAD_COUNT);
+
+        final CountDownLatch latch = new CountDownLatch(THREAD_COUNT);
+        for (int i = 0; i < THREAD_COUNT; i++) {
+            final int finalI = i;
+            execSer.execute(new Runnable() {
+                @Override
+                public void run() {
+                    HazelcastServerCachingProvider cachingProvider = createCachingProvider(hzConfig);
+                    Cache<Object, Object> cache = cachingProvider.getCacheManager().getCache("xmlCache"+finalI);
+                    cache.get(1);
+                    latch.countDown();
+                }
+            });
+        }
+        HazelcastTestSupport.assertOpenEventually(latch);
+        if (!execSer.awaitTermination(30, TimeUnit.SECONDS)) {
```

Comment:
Just put it in the @Before/@After methods of a test. I also see that you are doing 

assertEquals(0, HazelcastClient.getAllHazelcastClients().size());

This is also highly undesirable. Because it make a test difficult to understand and apparently you are not trusting something and you let this lack of confidence leak in your test. If could very well be that there is an issue; if so.. it needs to be addressed. Otherwise you should only rely on before/after methods to set your tests in a valid state.